h2o-core/src/main/java/hex/grid/GridSearch.java

```diff
@@ -98,6 +98,8 @@ private GridSearch(Key<Grid> gkey, HyperSpaceWalker<MP, ?> hyperSpaceWalker) {
       if (! (keyed instanceof Grid))
         throw new H2OIllegalArgumentException("Name conflict: tried to create a Grid using the ID of a non-Grid object that's already in H2O: " + _job._result + "; it is a: " + keyed.getClass());
       grid = (Grid) keyed;
+      // Old failures are no longer needed once new hyperspace search is issued by the user.
+      if ((grid.getFailureCount() > 0)) grid.initFailureFields();
```

Comment:
Hmm, is this a good idea? IMHO should keep the failures of the models that are part of this grid but discard them for the ones that are not part of this grid anymore, no?